checker/src/main/java/alluxio/checker/HiveIntegrationChecker.java

```diff
@@ -0,0 +1,202 @@
+        + "ROW FORMAT DELIMITED FIELDS TERMINATED BY '|' "
+        + "LOCATION '" + mAlluxioURL + "/alluxioTestFolder/'";
+    try (PreparedStatement createTablePS = con.prepareStatement(sql)) {
+      createTablePS.execute();
+    }
+  }
+
+  /**
+   * Supports the second integration way: use Alluxio as the default filesystem.
+   *
+   * @param con the Hive connection
+   * @param tableName the name of the test table
+   */
+  private void useAsUnderFS(Connection con, String tableName) throws Exception {
+    String sql = "CREATE TABLE " + tableName + " (ROW1 STRING, ROW2 STRING) "
+        + "ROW FORMAT DELIMITED FIELDS TERMINATED BY '|' "
+        + "STORED AS TEXTFILE";
+    try (PreparedStatement createTablePS = con.prepareStatement(sql)) {
+      createTablePS.execute();
+    }
+
+    sql = "INSERT INTO " + tableName + " VALUES ('Hive', 'Test')";
+    try (PreparedStatement loadTablePS = con.prepareStatement(sql)) {
+      loadTablePS.executeUpdate();
+    }
+  }
+
+  /**
+   * Translates the exception message to user-facing message.
+   *
+   * @param exception the thrown exception when running Hive queries
+   * @param reportWriter save user-facing messages to a generated file
+   */
+  private void printExceptionReport(Exception exception, PrintWriter reportWriter) {
+    String exceptionStr = exception.toString();
+    if (exceptionStr.contains("Class alluxio.hadoop.FileSystem not found")) {
+      reportWriter.println(FAIL_TO_FIND_CLASS_MESSAGE);
+      reportWriter.println(TEST_FAILED_MESSAGE);
+    } else if (exceptionStr.contains("No FileSystem for scheme \"alluxio\"")
+        || exceptionStr.contains("No FileSystem for scheme: alluxio")) {
+      reportWriter.println(FAIL_TO_FIND_FS_MESSAGE);
+      reportWriter.println(TEST_FAILED_MESSAGE);
+    } else {
+      reportWriter.println("Other errors occur, please fix it and "
+          + "rerun the Hive Integration Checker. \n");
+      exception.printStackTrace(reportWriter);
+      reportWriter.println(TEST_FAILED_MESSAGE);
+    }
+  }
+
+  /**
+   * Checks if the input arguments are valid.
+   */
+  private void checkIfInputValid() throws IllegalArgumentException {
+    if (mUserMode != 1 && mUserMode != 2) {
+      throw new IllegalArgumentException("Hive integration checker user mode is invalid.");
+    }
+    if (!mHiveURL.startsWith("jdbc")) {
+      throw new IllegalArgumentException("Hive URL is invalid.");
+    }
+  }
+
+  /**
+   * Main function will be triggered by hive-checker.sh.
+   *
+   * @param args the arguments to connect to Hive and define the test mode
+   */
+  public static void main(String[] args) throws Exception {
+    HiveIntegrationChecker checker = new HiveIntegrationChecker();
+    JCommander jCommander = new JCommander(checker, args);
+    jCommander.setProgramName("HiveIntegrationChecker");
+    try {
+      checker.checkIfInputValid();
+    } catch (IllegalArgumentException e) {
+      System.out.println(e);
```

Comment:
does jcommand provide usage utils to print out? this can be a good location to do so